### PR TITLE
improve(planner): utilize with `HashMethodFixedKeys` in hash join

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

As the title

## Changelog

- Improvement

## Related Issues

Fixes https://github.com/datafuselabs/databend/issues/5598

### CODE REVIEW

File: query/src/pipelines/new/processors/transforms/hash_join/chaining_hash_table.rs [hunk 1]
```diff
@@ -12,28 +12,42 @@
...
 use crate::common::HashMap;
+use crate::common::HashTableKeyable;
+use crate::pipelines::new::processors::transforms::hash_join::row::Chunk;
 use crate::pipelines::new::processors::transforms::hash_join::row::RowPtr;
 use crate::pipelines::new::processors::transforms::hash_join::row::RowSpace;
 use crate::pipelines::new::processors::HashJoinState;
 use crate::pipelines::transforms::group_by::keys_ref::KeysRef;
 use crate::sessions::QueryContext;
 
+pub enum HashTable {
```
Comment (leiysky) [inline query/src/pipelines/new/processors/transforms/hash_join/chaining_hash_table.rs:43]: Maybe you can put the `HashMethod` here?

**Comment by sundy-li** [comment]
Maybe it's better to merge after #5678  is finished.

File: query/src/pipelines/new/processors/transforms/hash_join/chaining_hash_table.rs (hunk 1 above)
Comment (xudong963) [reply query/src/pipelines/new/processors/transforms/hash_join/chaining_hash_table.rs:43]: Do you mean
```rust
pub enum HashTable {
    SerializerHashTable(HashMethodSerializer),
    KeyU8HashTable(HashMethodFixedKeys<u8>),
    ...
}
```

**Comment by BohuTANG** [comment]
> Maybe it's better to merge after #5678 is finished.

5678(a good PR number) has merged!

**Comment by xudong963** [comment]
> > Maybe it's better to merge after #5678 is finished.
> 
> 5678(a good PR number) has merged!

I'm adding new HashMethods

File: query/src/pipelines/new/processors/transforms/hash_join/chaining_hash_table.rs (hunk 1 above)
Comment (leiysky) [reply query/src/pipelines/new/processors/transforms/hash_join/chaining_hash_table.rs:45]: > Do you mean
> 
> ```rust
> 
> pub enum HashTable {
> 
>     SerializerHashTable(HashMethodSerializer),
> 
>     KeyU8HashTable(HashMethodFixedKeys<u8>),
> 
>     ...
> 
> }
> 
> ```

Yes, it's better to associate the structs with same generic types together.

**Review by sundy-li — APPROVED** [review]

**Pull request merged**